delivery_product_restriction/models/delivery_carrier.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 Coop IT Easy SCRLfs
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class DeliveryCarrier(models.Model):
+
+    _inherit = "delivery.carrier"
+
+    def _can_be_used_to_deliver_products(self, products):
+        self.ensure_one()
+        return all(
+            not product.restrict_delivery_carrier_to
+            or self in product.restrict_delivery_carrier_to
+            for product in products
+        )
+
+    def available_carriers(self, partner, products=None):
```

Comment:
Since you do not call super, it's not obvious this is an overloaded function. So I'd suggest leaving a comment :-)

Since it's an overload, it's totally ok to leave the name as is and the function non static. 👍

Reply:
I will add a comment. :)